network/src/protocols/identify/protocol.rs

```diff
@@ -69,7 +70,12 @@ impl<'a> IdentifyMessage<'a> {
             Multiaddr::try_from(reader.observed_addr().bytes().raw_data().to_vec()).ok()?;
         let mut listen_addrs = Vec::with_capacity(reader.listen_addrs().len());
         for addr in reader.listen_addrs().iter() {
-            listen_addrs.push(Multiaddr::try_from(addr.bytes().raw_data().to_vec()).ok()?)
+            match Multiaddr::try_from(addr.bytes().raw_data().to_vec()) {
+                Ok(multi_addr) => {
+                    listen_addrs.push(multi_addr);
```

Comment:
any specific reason for ignore this error?

Reply:
because `reader.listen_addrs()` have many addresses. In `reader.listen_addrs().iter()`, if we only parsed a addressed failed, there maybe others addresses are ok.

@driftluo  what's your opinion?